source/main/gui/panels/GUI_GameMainMenu.cpp

```diff
@@ -84,58 +108,97 @@ void GameMainMenu::DrawMenuPanel()
+    if (ImGui::Begin(_LC("MainMenu", title), nullptr, static_cast<ImGuiWindowFlags_>(flags)))
     {
         int button_index = 0;
         ImVec2 btn_size(WINDOW_WIDTH, 0.f);
-
-        if (HighlightButton(_LC("MainMenu", "Single player"), btn_size, button_index++))
+        if (App::app_state->GetEnum<AppState>() == AppState::MAIN_MENU)
         {
-            this->SetVisible(false);
-            if (App::diag_preset_terrain->GetStr().empty())
-            {
-                RoR::Message m(MSG_GUI_OPEN_SELECTOR_REQUESTED);
-                m.payload = reinterpret_cast<void*>(new LoaderType(LT_Terrain));
-                App::GetGameContext()->PushMessage(m);
-            }
-            else
+            if (HighlightButton(_LC("MainMenu", "Single player"), btn_size, button_index++))
             {
-                App::GetGameContext()->PushMessage(Message(MSG_SIM_LOAD_TERRN_REQUESTED, App::diag_preset_terrain->GetStr()));
+                this->SetVisible(false);
+                if (App::diag_preset_terrain->GetStr().empty())
+                {
+                    RoR::Message m(MSG_GUI_OPEN_SELECTOR_REQUESTED);
+                    m.payload = reinterpret_cast<void*>(new LoaderType(LT_Terrain));
+                    App::GetGameContext()->PushMessage(m);
+                }
+                else
+                {
+                    App::GetGameContext()->PushMessage(Message(MSG_SIM_LOAD_TERRN_REQUESTED, App::diag_preset_terrain->GetStr()));
+                }
             }
         }
 
         if (FileExists(PathCombine(App::sys_savegames_dir->GetStr(), "autosave.sav")))
         {
             if ( HighlightButton(_LC("MainMenu", "Resume game"), btn_size, button_index++))
             {
-                App::GetGameContext()->PushMessage(Message(MSG_SIM_LOAD_SAVEGAME_REQUESTED, "autosave.sav"));
+                if (App::app_state->GetEnum<AppState>() == AppState::MAIN_MENU)
+                {
+                    App::GetGameContext()->PushMessage(Message(MSG_SIM_LOAD_SAVEGAME_REQUESTED, "autosave.sav"));
+                }
+                else
+                {
+                    App::GetGameContext()->PushMessage(Message(MSG_SIM_UNPAUSE_REQUESTED));
+                }
                 this->SetVisible(false);
             }
         }
 
-        if (HighlightButton(_LC("MainMenu", "Multiplayer"), btn_size, button_index++))
+        if (App::app_state->GetEnum<AppState>() != AppState::MAIN_MENU && App::mp_state->GetEnum<MpState>() != MpState::CONNECTED)
         {
-            App::GetGuiManager()->SetVisible_MultiplayerSelector(true);
-            this->SetVisible(false);
+            if ( HighlightButton(_LC("MainMenu", "Change map"), btn_size, button_index++))
+            {
+                if (App::diag_preset_terrain->GetStr().empty())
+                {
+                    this->SetVisible(false);
+                    App::GetGameContext()->PushMessage(Message(MSG_SIM_UNLOAD_TERRN_REQUESTED));
```

Comment:
this is correct but not what I had in mind. I'd like the map selector to be displayed on top of the simulation and only send UNLOAD message if user presses OK. also, we may reuse the "single player" button, see the other comment.